src/main/java/javaslang/collection/Vector.java

```diff
@@ -1342,9 +1342,9 @@ private Object readResolve() {
     public boolean equals(Object o) {
         if (o == this) {
             return true;
-        } else if (o instanceof Vector) {
+        } else if (o instanceof IndexedSeq) {
```

Comment:
Hi Ruslan, what was your intention here? Changing it to super-interface IndexedSeq should work. This is a general philosophy: when are objects equal.

To keep it consistent with the other classes I would keep `Vector` here.